MOBILE_CHECKOUT_MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,190 @@
+    +        // Handle cancel case
+    +    }
+
+    +    func paypalWillStart(_ payPalClient: PayPalNativePayments.PayPalNativeCheckoutClient) {
+    +        // Handle any UI clean-up before paysheet presents
+    +    }
+    + }
+    ```
+    
+6. Implement shipping delegate
+
+    :warning: Only implement `PayPalNativeShippingDelegate` if your order ID was created with [`shipping_preference`](https://developer.paypal.com/docs/api/orders/v2/#definition-experience_context_base) = `GET_FROM_FILE`. If you created your order ID with `shipping_preference` = `NO_SHIPPING` or `SET_PROVIDED_ADDRESS`, **skip this step** (step 6).
+
+
+    * `PayPalNativeShippingDelegate` notifies your app when the user updates their shipping address **or** shipping method. 
+        * In the previous SDK, both shipping change types were lumped into one `ShippingChangeCallback`.
+    * You are required to PATCH the order details on your server if the shipping method (or amount) changes. Do this with the [PayPal Orders API - Update order](https://developer.paypal.com/docs/api/orders/v2/#orders_patch) functionality.
+
+    ```diff=
+    + extension ViewController: PayPalNativeShippingDelegate {
+
+    +    func paypal(
+    +        _ payPalClient: PayPalNativeCheckoutClient,
+    +        didShippingAddressChange shippingAddress: PayPalNativeShippingAddress,
+    +        withAction shippingActions: PayPalNativePaysheetActions
+    +    ) {
+    +       // called when the user updates their chosen shipping address
+    
+    +       // REQUIRED: you must call actions.approve() or actions.reject() in this callback
+    +       actions.approve()
+    
+    +       // OPTIONAL: you can optionally patch your order. Once complete, call actions.approve() if successful or actions.reject() if not.
+    +    }
+    
+    +    func paypal(
+    +        _ payPalClient: PayPalNativeCheckoutClient,
+    +        didShippingMethodChange shippingMethod: PayPalNativeShippingMethod,
+    +        withAction shippingActions: PayPalNativePaysheetActions
+    +    ) {
+    +        // Handle user shipping method selection change
+    +
+    +        // REQUIRED: patch your order server-side with the updated shipping amount
+    +        // Once complete, call `actions.approve()` or `actions.reject()` 
+    +        if patchOrder() == .success {
+    +            actions.approve()
+    +        } else {
+    +            actions.reject()
+    +        }
+    +    }
+    + }
+    ```
+
+7. Remove the old SDK dependency
+
+* Swift Package Manager
+    * Remove the `paypalcheckout-ios` package from your Xcode project's "Package Dependencies"
+    * ![](https://iili.io/HO1okCB.png)
+
+* CocoaPods
+    * Remove `pod 'PayPalCheckout'` from your Podfile
+    * Refresh your local `/Pods`
+ 
+## Resources
+
+We migrated a sample app for you to reference:
+
+* [iOS PR](https://github.com/scannillo/paypal-xo-sample/pull/1): ➕101 lines ➖39 lines
```

Comment:
Should we remove this section, since Android's isn't complete yet? It's definitely a nice-to-have, so I figured I would leave it around for iOS. It might be a pain to have to maintain though. So maybe just safer to omit?

I will say - doing the sample app while writing this guide was a great reference. It definitely pointed out holes in the process!

Reply:
I would vote leave it for now. May be worth calling out versions somewhere, like the current version of MXO and PPCP. That way if things change it's at least documented the version of each SDK the migration was on. 

Maybe we just throw a ticket somewhere in the backlog to remove this after a certain number of months. That way people aren't asking why it's outdated a year from now. I do think it's helpful for the initial migration though.

Reply:
That's a great call. I'll update that sample PR description to include the SDK versions!